### DISCUSSION TITLE
Do Promise.allSettled trigger catch block in any case?

### DISCUSSION THREAD

**Discussion by pugazhendhis** [mdn/mdn-community#551] [Content]
Example code:

```
const promise1 = () => Promise.resolve('promise1');

const promise2 = () => Promise.reject('promise1');

Promise.allSettled([promise1(), promise2()]).then(()=> 'success').catch((error) => 'failure');

```
I know that `Promise.allsettled` returns array of settled (resolve/reject) Promise.

But above snippet returns `success` from then callback though `promise2 rejects`.

My question is in which scenario `failure` from catch callback will trigger?

**Comment by Josh-Cena** [comment]
It doesn't in most conceivable cases, but just for the fun of it, you can indeed trigger rejections in some corner cases:

```js
// An iterable that throws when iterated
Promise.allSettled({ get [Symbol.iterator]() { throw new Error("No iterator"); } }).catch(console.log);

Promise.allSettled((function* () { throw new Error("Can't iterate me"); })()).catch(console.log);

// Calling allSettled on an incompatible constructor
Promise.allSettled.call(Number).catch(console.log); // TypeError: Promise resolve or reject function is not callable
```

...but in general you don't have to worry about that and I find it too obscure to be documented.
